sdcm/cluster.py

```diff
@@ -3045,22 +3048,47 @@ def kill_gemini_thread(self):
                 if kill_result.exit_status != 0:
                     self.log.error('Terminate gemini on node %s:\n%s', loader, kill_result)
 
+    def get_ks_cf_list(self, loader_node, db_node, request_timeout=300, collect_system=False):
+        cmd = "paging off; SELECT keyspace_name, table_name, type FROM system_schema.columns"
+        result = loader_node.run_cqlsh(cmd=cmd, timeout=request_timeout, verbose=False, target_db_node=db_node,
+                                       split=True, connect_timeout=request_timeout)
+
+        avaialable_ks_cf = defaultdict(list)
+        for row in result[4:]:
+            if '|' not in row:
+                continue
+            row_splitted = row.split('|')
+            ks_cf_name = '.'.join([name.strip() for name in row_splitted[:2]])
+            if not collect_system and ks_cf_name.startswith('system'):
+                continue
+            column_type = row_splitted[2].strip()
+            avaialable_ks_cf[ks_cf_name].append(column_type)
+        return avaialable_ks_cf
+
     def get_non_system_ks_cf_list(self, loader_node, db_node, request_timeout=300):
         """Get all not system keyspace.tables pairs
 
         Arguments:
             loader_node {BaseNode} -- LoaderNoder to send request
             db_node_ip {str} -- ip of db_node
         """
+        avaialable_ks_cf = self.get_ks_cf_list(loader_node, db_node, request_timeout=request_timeout)
```

Comment:
I don't understand why we need this function if the above function already does it...
Either you leave only one of them or you have the new function above to retrieve all ks.cf and this one will filter only the non-system.